src/resource_files.rs

```diff
@@ -95,6 +106,17 @@ impl Deref for ResourceFiles {
     }
 }
 
+impl Guard for ResourceFiles {
+    fn check(&self, ctx: &GuardContext<'_>) -> bool {
+        for filename in self.inner.files.keys() {
```

Comment:
I think this can be replaced with `self.inner.files.contains_key(ctx.head().uri.path().trim_start_matches('/'))`

Other things I figured out looking at this code:

1. We should not create same object for guarded variant, presence of this guard means we do not need to have complicated logic to determine if file is there: we already know it is there 100%. Other checks like index.html look ups and etc. makes no sense, if they already denied by guard. Our API must not allow misusage (for example, guarded variant cannot have `resolve_defaults` option).
2. I have feeling we should not clone ourself and implement the Guard trait, but maybe we should separate structure dedicated only to this?
3. If we call this option `use_guard`, we tell user only we will use internals of `actix-web` to do something unclear, he had to read futher documentation. Maybe we should name this in a way, thay use will understand purpose of method without looking into docs? Like `use_next_route_if_not_found`? Maybe a little bit long :)

Reply:
Thanks! I'll update PR.

Reply:
- I've used your suggestion to make the comparison faster.
- We no longer clone the file contents, only the filenames (also faster)
- Renamed function to `skip_handler_when_not_found`

Let me know if you need other changes.

Reply:
I think I will merge your changes with this variant first and then will do some extra steps to disallow API misusage.

Reply:
Awesome! Can you publish these changes? I can no longer `cargo publish` because I'm using my github fork. Thanks in advance1

Reply:
Yeap, will do it soon, most probably over weekend, have pretty busy week as usual :)

Reply:
@joepio I've updated implementation in master. Now it supports also defaults resolution.

Could you try this variant and say if it works for you? You need to update you routes.rs like this:

```
generate()).skip_handler_when_not_found().do_not_resolve_defaults()
```
If everything will be fine - I will publish crate.

Reply:
Seems to work great here! :)